src/actions/swapParent.ts

```diff
@@ -20,25 +22,67 @@ const swapParent = (state: State) => {
   // If the cursor is at the root, do nothing.
   if (!parent.length) return state
 
-  const childValue = getThoughtById(state, head(cursor))?.value
-  const parentValue = getThoughtById(state, head(parent))?.value
-  if (!childValue || !parentValue) return state
+  const childId = head(cursor)
+  const parentId = head(parent)
 
-  const reducers = [
-    editThought({
-      oldValue: childValue,
-      newValue: parentValue,
-      path: simplifyPath(state, cursor),
-      force: true,
-    }),
-    editThought({
-      oldValue: parentValue,
-      newValue: childValue,
-      path: simplifyPath(state, parent),
-    }),
-  ]
+  const childThought = getThoughtById(state, childId)
+  const parentThought = getThoughtById(state, parentId)
+  if (!childThought || !parentThought) return state
+
+  // Get only direct children of the child thought (grandchildren)
+  const childChildren = getChildrenRanked(state, childId)
+
+  // Get the grandparent path (parent of parent)
+  const grandparent = parentOf(parent)
+
+  // Get siblings (other children of parent excluding the child being swapped)
+  const parentChildren = getChildrenRanked(state, parentId)
+  const siblings = parentChildren.filter(sibling => sibling.id !== childId)
 
-  return reducerFlow(reducers)(state)
+  return reducerFlow([
+    // First move the child to replace its parent's position
+    state =>
```

Comment:
Currying really shines in situations like this where `state` is passed through the flow functions, but is not needed for the arguments.

Reply:
Thanks, that makes sense. I'll keep that in mind.